dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java

```diff
@@ -0,0 +1,209 @@
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.business.HostAPI;
+import com.dotmarketing.portlets.htmlpageasset.business.HTMLPageAssetAPI;
+import com.dotmarketing.portlets.htmlpageasset.business.render.page.HTMLPageAssetRendered;
+import com.dotmarketing.portlets.htmlpageasset.business.render.page.HTMLPageAssetRenderedBuilder;
+import com.dotmarketing.portlets.htmlpageasset.business.render.page.PageView;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.languagesmanager.business.LanguageAPI;
+import com.dotmarketing.portlets.templates.business.TemplateAPI;
+import com.dotmarketing.portlets.templates.design.bean.TemplateLayout;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.*;
+import com.liferay.portal.model.User;
+import org.apache.velocity.context.Context;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * {@link HTMLPageAssetRenderedAPI} implementation
+ */
+public class HTMLPageAssetRenderedAPIImpl implements HTMLPageAssetRenderedAPI {
+
+    private final HostWebAPI hostWebAPI = WebAPILocator.getHostWebAPI();
+    private final HTMLPageAssetAPI htmlPageAssetAPI = APILocator.getHTMLPageAssetAPI();
+    private final LanguageAPI languageAPI = APILocator.getLanguageAPI();
+    private final VersionableAPI versionableAPI = APILocator.getVersionableAPI();
+    private final TemplateAPI templateAPI = APILocator.getTemplateAPI();
+    private final HostAPI hostAPI = APILocator.getHostAPI();
+    private final PermissionAPI permissionAPI = APILocator.getPermissionAPI();
+    private final UserAPI userAPI = APILocator.getUserAPI();
+    private final ContainerAPI containerAPI = APILocator.getContainerAPI();
+    private final ContainerRenderedBuilder containerRenderedBuilder = ContainerRenderedBuilder.get();
+
+    @Override
+    public HTMLPageAssetRendered getPageRendered(HttpServletRequest request, HttpServletResponse response, User user,
+                                               String pageUri, PageMode pageMode)
+            throws DotDataException, DotSecurityException, IOException {
+
+        final HTMLPageAsset page = this.getPage(request, user, pageUri, pageMode);
+
+        if (page == null) {
+            throw new HTMLPageAssetNotFoundException(pageUri);
+        }
+
+        return this.getPageRendered(request, response, user, page, pageMode);
+    }
+
+    @Override
+    public HTMLPageAssetRendered getPageRendered(final HttpServletRequest request, final HttpServletResponse response,
+                                                 final User user, final HTMLPageAsset page, PageMode pageMode)
+            throws DotDataException, DotSecurityException, IOException {
+
+        return HTMLPageAssetRenderedBuilder.get()
```

Comment:
this is not the right way to do the builder, it should be a separated class such as
new HTMLPageAssetRenderedBuilder().htmlPageAsset(page).user(user).request(request).....build()